db/system_keyspace.hh

```diff
@@ -285,7 +285,6 @@ public:
     future<> make(
             locator::effective_replication_map_factory&,
             replica::database&,
-            db::config&,
```

Comment:
I know how it happened that this arg became unused.
<boring mode>
The db::config is the generic scylla-wide configuration container that used to sit on a database object from day-0. And everybody got used to the fact that they can get this config out of database. Eventually, the intention became that every service should have its own specialized config, e.g. database got its `database_config` thing, so did many other services. The db::config was supposed to have its scope limited to main.cc (and some of its pals) and be used to generate services' configs. Here in system_keyspace::make() the db::config was sort of used like that, but at some point the database& argument appeared and instead of db::config& cfg the db.get_config() took its place.
</boring mode>